### PR TITLE
feat(rds): validate backup retention for read replica instances

### LINKED ISSUE #17356: (aws-rds): DB Backups not supported on a read replica for engine postgres
### General Issue

based on #16934 

### The Question

This issue-based on #16934 

I just want to upgrade my instance type from `instanceType: ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.SMALL)` to `instanceType: ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.LARGE)`, and not workable.

probably if I want to change to other different instance types, it maybe will show the same error messages too.

### CDK CLI Version

1.130.0 (build 9c094ae)

### Framework Version

1.130.0 (build 9c094ae)

### Node.js Version

v14.18.0

### OS

Ubuntu 20.04.3 LTS

### Language

Typescript

### Language Version

typescript@3.9.10

### PR DESCRIPTION
**Pull request by jogold**
Automatic backups of read replica instances are only supported for MySQL
and MariaDB.

See https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_ReadRepl.html#USER_ReadRepl.Overview.Differences

Closes #17356

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-rds/lib/instance.ts [hunk 1]
```diff
@@ -1143,6 +1143,12 @@ export class DatabaseInstanceReadReplica extends DatabaseInstanceNew implements
   constructor(scope: Construct, id: string, props: DatabaseInstanceReadReplicaProps) {
     super(scope, id, props);
 
+    if (props.sourceDatabaseInstance.engine
+        && !props.sourceDatabaseInstance.engine.supportsReadReplicaBackups
+        && props.backupRetention) {
+      throw new Error(`Cannot set \`backupRetention\` for a ${props.sourceDatabaseInstance.engine.engineType} read replica. Automatic backups are not supported.`);
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/instance.ts:1149]: We actually have a function for this - `engineDescription()` (it's in `lib/private/util.ts`).

File: packages/@aws-cdk/aws-rds/test/instance.test.ts [hunk 2]
```diff
@@ -1583,6 +1583,24 @@ describe('instance', () => {
...
+      instanceType,
+      vpc,
+    });
+
+    expect(() => new rds.DatabaseInstanceReadReplica(stack, 'Replica', {
+      sourceDatabaseInstance: source,
+      backupRetention,
+      instanceType,
+      vpc,
+    })).toThrow(/Cannot set `backupRetention` for a postgres read replica/);
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/test/instance.test.ts:1602]:
```suggestion
    expect(() =>{
      new rds.DatabaseInstanceReadReplica(stack, 'Replica', {
        sourceDatabaseInstance: source,
        backupRetention,
        instanceType,
        vpc,
      });
    }).toThrow(/Cannot set `backupRetention` for a postgres read replica/);
```

(Sorry, GutHub's UI seems screwed up today, I can't select lines correctly)

File: packages/@aws-cdk/aws-rds/lib/instance.ts [hunk 3]
```diff
@@ -1143,6 +1143,12 @@ export class DatabaseInstanceReadReplica extends DatabaseInstanceNew implements
...
+    }
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/instance.ts:1150]: The error message seems a little contrived... how about:

```suggestion
      throw new Error(`Cannot set 'backupRetention', as engine '${engineDescription(engine)}' does not support automatic backups for read replicas`);
```

**Review by skinny85 — CHANGES_REQUESTED** [review]
Looks great! Tiny comments, mainly about using a private function we have in the module (`engineDescription()`).

File: packages/@aws-cdk/aws-rds/test/instance.test.ts (hunk 2 above)
Comment (jogold) [reply packages/@aws-cdk/aws-rds/test/instance.test.ts]: Done. A note on this `{ }` style for `toThrow` expectations: it allows to "easily" add statements inside the `{ }` which is not a good practice when testing that a single statement should throw.

```ts
expect(() => myFun()).toThrow(); // cannot easily add statements inside the throw expectation
```

```ts
expect(() => {
  myFun(); // can easily add statements inside (above or below)
}).toThrow();
```

😄

File: packages/@aws-cdk/aws-rds/README.md [hunk 4]
```diff
@@ -129,6 +129,10 @@ new rds.DatabaseInstanceReadReplica(this, 'ReadReplica', {
 });
 ```
 
+Automatic backups of read replica instances is only supported for MySQL and MariaDB. By default,
```
Comment (jogold) [inline packages/@aws-cdk/aws-rds/README.md:132]:
```suggestion
Automatic backups of read replica instances are only supported for MySQL and MariaDB. By default,
```

File: packages/@aws-cdk/aws-rds/test/instance.test.ts (hunk 2 above)
Comment (skinny85) [reply packages/@aws-cdk/aws-rds/test/instance.test.ts]: You are correct, but for me, it's more important that I see what's being tested o(`myFun()` in this case) on a separate line 🙂.

Comment (jogold) [reply packages/@aws-cdk/aws-rds/test/instance.test.ts]: OK. The change has been made.

**Review by skinny85 — APPROVED** [review]
Amazing work, as usual. Thanks @jogold!

**Pull request merged**